LiskHQ/lisk-sdk
# Create protocol-spec for all fork choice patterns - Closes #4129

### What was the problem?
Fork choice rules were missing protocol specifications and the tests.
### How did I solve it?
Covered fork choice scenarios with integration tests, generating the blocks as inputs and passing them to the block processing mechanism. I didn't generate the .json input blocks beforehand, but rather generating the blocks as part of the regular scenario.

### How to manually test it?
Run tests, query Blocks table to see if the last applied block is matching the expected results.
### Review checklist

- [ ] The PR resolves #4129
